### PR TITLE
E2e tests for 1.24 Ubuntu vsphere

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
#3491 

*Description of changes:*
E2E tests for vsphere 1.24 Ubuntu

*Testing (if applicable):*
- make e2e

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/resource/reconciler_test.go [hunk 1]
```diff
@@ -894,7 +894,6 @@ func TestClusterReconcilerReconcileCloudStack(t *testing.T) {
 					},
 				}
 
-
```
Comment (mitalipaygude) [inline controllers/resource/reconciler_test.go:897]: Formatting changes

File: test/e2e/awsiam_test.go [hunk 2]
```diff
@@ -227,8 +257,8 @@ func TestSnowKubernetes121UbuntuAWSIamAuth(t *testing.T) {
 		framework.NewSnow(t, framework.WithSnowUbuntu121()),
 		framework.WithAWSIam(),
 		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube121)),
-		framework.WithEnvVar("SNOW_PROVIDER", "true"),
-		framework.WithEnvVar("FULL_LIFECYCLE_API", "true"),
```
Comment (vivek-koppuru) [inline test/e2e/awsiam_test.go:231]: We don't need this env var anymore?

File: test/e2e/upgrade_test.go [hunk 3]
```diff
@@ -114,43 +114,45 @@ func TestVSphereKubernetes123UbuntuTo124Upgrade(t *testing.T) {
...
 
-func TestVSphereKubernetes122UbuntuTo123MultipleFieldsUpgrade(t *testing.T) {
-	provider := framework.NewVSphere(t, framework.WithUbuntu122())
+func TestVSphereKubernetes123UbuntuTo124MultipleFieldsUpgrade(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu123())
 	test := framework.NewClusterE2ETest(
 		t,
 		provider,
-		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube122)),
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube123)),
```
Comment (vivek-koppuru) [inline test/e2e/upgrade_test.go:148]: You didn't add the env var for 1.24 here, but I don't think it is needed

File: test/e2e/vsphere_disable_csi_test.go [hunk 4]
```diff
@@ -41,3 +42,20 @@ func TestVSphereKubernetesDisableCSIUpgrade(t *testing.T) {
 		provider,
 	)
 }
+
+func TestVSphereKubernetes124DisableCSIUpgrade(t *testing.T) {
```
Comment (vivek-koppuru) [inline test/e2e/vsphere_disable_csi_test.go:46]: We don't need 1.24 test here. Just replace the above one.

File: test/e2e/awsiam_test.go (hunk 2 above)
Comment (mitalipaygude) [reply test/e2e/awsiam_test.go:231]: we still need it, just using the features package to use the constant instead.

File: test/e2e/conformance_test.go [hunk 5]
```diff
@@ -110,6 +111,17 @@ func TestVSphereKubernetes123ThreeWorkersConformanceFlow(t *testing.T) {
 	runConformanceFlow(test)
 }
 
+func TestVSphereKubernetes124ThreeWorkersConformanceFlow(t *testing.T) {
+	test := framework.NewClusterE2ETest(
+		t,
+		framework.NewVSphere(t, framework.WithUbuntu123()),
```
Comment (vivek-koppuru) [inline test/e2e/conformance_test.go:117]:
```suggestion
		framework.NewVSphere(t, framework.WithUbuntu124()),
```

**Review by vivek-koppuru — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**